tests/area/test_markets.py

```diff
@@ -0,0 +1,132 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+
+from unittest.mock import Mock, patch
+
+import pytest
+from d3a.d3a_core.device_registry import DeviceRegistry
+from d3a.models.area import Area
+from d3a.models.strategy.storage import StorageStrategy
+from d3a_interface.constants_limits import ConstSettings, TIME_ZONE
+from pendulum import duration, today
+
+
+class TestMarketRotation:
+
+    @pytest.fixture
+    def area_fixture(self):
+        original_registry = DeviceRegistry.REGISTRY
+        DeviceRegistry.REGISTRY = {
+            "General Load": (33, 35),
+        }
+        config = Mock()
+        config.slot_length = duration(minutes=15)
+        config.tick_length = duration(seconds=15)
+        config.ticks_per_slot = 60
+        config.start_date = today(tz=TIME_ZONE)
+        config.grid_fee_type = ConstSettings.IAASettings.GRID_FEE_TYPE
+        config.end_date = config.start_date + duration(days=1)
+        config.market_count = 5
+        child = Area(name="test_market_area", config=config, strategy=StorageStrategy())
+        area = Area(name="parent_area", children=[child], config=config)
+        yield area
+
+        DeviceRegistry.REGISTRY = original_registry
+
+    def test_market_rotation_is_successful(self, area_fixture):
+        area_fixture.activate()
+        assert len(area_fixture.all_markets) == 5
+        slot_length_in_ticks = area_fixture.config.slot_length / area_fixture.config.tick_length
+        area_fixture.current_tick = slot_length_in_ticks
+        area_fixture.cycle_markets()
+        assert len(area_fixture.past_markets) == 1
+        assert len(area_fixture.all_markets) == 5
+
+        area_fixture.current_tick = slot_length_in_ticks * 2
+        area_fixture.cycle_markets()
+        assert len(area_fixture.past_markets) == 1
+        assert len(area_fixture.all_markets) == 5
+
+    @patch("d3a.constants.RETAIN_PAST_MARKET_STRATEGIES_STATE", True)
+    def test_market_rotation_is_successful_keep_past_markets(self, area_fixture):
+        area_fixture.activate()
+        assert len(area_fixture.all_markets) == 5
+        slot_length_in_ticks = area_fixture.config.slot_length / area_fixture.config.tick_length
```

Comment:
IMO this is more understandable
```suggestion
        ticks_per_slot = area_fixture.config.slot_length / area_fixture.config.tick_length
```